diff --git a/libvmaf/src/meson.build b/libvmaf/src/meson.build
index a3e1d414..8c42d950 100644
--- a/libvmaf/src/meson.build
+++ b/libvmaf/src/meson.build
@@ -399,6 +399,28 @@ if is_cuda_enabled
             gencode += '-gencode=arch=compute_80,code=compute_80'
         endif
 
+        # =============|==============|=================
+        # Generation   | Compute Code | Typical Card
+        # -------------|--------------|-----------------
+        # Maxwell      | 5.0          | GeForce GTX 750
+        # Maxwell      | 5.2          | GeForce GTX 980
+        # Pascal       | 6.1          | GeForce GTX 1080
+        # Turing       | 7.5          | GeForce RTX 2080
+        # Ampere       | 8.6          | GeForce RTX 3080
+        # Ada Lovelace | 8.9          | GeForce RTX 4090
+        # Blackwell    | 12.0         | GeForce RTX 5090
+        # =============|==============|=================
+        gencode = [
+            '--fatbin',
+            '-gencode=arch=compute_120,code=compute_120',
+            '-gencode=arch=compute_120,code=sm_120',
+            '-gencode=arch=compute_89,code=sm_89',
+            '-gencode=arch=compute_86,code=sm_86',
+            '-gencode=arch=compute_80,code=compute_80',
+            '-gencode=arch=compute_75,code=sm_75',
+            '-O3'
+        ]
+
     else
         nvcc_exe = find_program('clang')
         gencode = [
